resources/file_attrs.go

```diff
@@ -0,0 +1,44 @@
+// Mgmt
+// Copyright (C) 2013-2016+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// +build go1.7
+
+package resources
+
+import (
+	"fmt"
+	"os/user"
+	"strconv"
+)
+
+func (obj *FileRes) gid(defaultVal int) (int, error) {
```

Comment:
Ok, i'll add: gid returns the group id specified in the graph or defaultVal if no value is specified. This function resolves the value in the graph to a numerical id for the current platform.